ui/components/app/basic-configuration-modal/basic-configuration-modal.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  Display,
+  FlexDirection,
+  AlignItems,
+  JustifyContent,
+  TextVariant,
+  BlockSize,
+  IconColor,
+  FontWeight,
+} from '../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { toggleExternalServices } from '../../../store/actions';
+import {
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  Modal,
+  Box,
+  Text,
+  ModalFooter,
+  Button,
+  IconName,
+  ButtonVariant,
+  Icon,
+  IconSize,
+  Checkbox,
+  ButtonSize,
+  Label,
+} from '../../component-library';
+import { getUseExternalServices } from '../../../selectors';
+import {
+  hideBasicFunctionalityModal,
+  onboardingToggleBasicFunctionalityOff,
+} from '../../../ducks/app/app';
+import { ONBOARDING_PRIVACY_SETTINGS_ROUTE } from '../../../helpers/constants/routes';
+
+export function BasicConfigurationModal() {
+  const t = useI18nContext();
+  const [hasAgreed, setHasAgreed] = useState(false);
+  const dispatch = useDispatch();
+  const isExternalServicesEnabled = useSelector(getUseExternalServices);
+  const { pathname } = useLocation();
+  const onboardingFlow = pathname === ONBOARDING_PRIVACY_SETTINGS_ROUTE;
+
+  function closeModal() {
+    dispatch(hideBasicFunctionalityModal());
+  }
+
+  return (
+    <Modal
+      onClose={closeModal}
+      data-testid="dapp-permission-modal"
+      isOpen={true}
```

Comment:
Nit: can be refactored as 
```suggestion
      isOpen
```